server/src/main/scala/services/graphstages/ArrivalsGraphStage.scala

```diff
@@ -260,23 +260,97 @@ class ArrivalsGraphStage(name: String = "",
       mergeBestFieldsFromSources(baseArrival, mergeArrival(key).getOrElse(baseArrival))
     }
 
+
+    def arrivalMapForSourceType(searchSourceType: ArrivalsSourceType): SortedMap[UniqueArrival, Arrival] = searchSourceType match {
+      case LiveArrivals => liveArrivals
+      case LiveBaseArrivals => liveBaseArrivals
+      case BaseArrivals => forecastBaseArrivals
+    }
+
+    def potentialKeys(mergeKey: UniqueArrival, searchArrivalMap: SortedMap[UniqueArrival, Arrival]): Iterable[UniqueArrival] =
+      searchArrivalMap.keys.filter(_.potentialKey(mergeKey, 1 * 60 * 60 * 1000))
+
+    def mergeScheduledApproximation(mergeKey: UniqueArrival, mergeKeySourceType: ArrivalsSourceType, foundArrival: Arrival, searchSourceType: ArrivalsSourceType) = {
```

Comment:
Could we think of a clearer name than `foundArrival`? When looking at this function it's not clear to me what it means. Would `arrivalToMerge` make sense?

Reply:
merge when we found Scheduled Approximation  for given mergeKey .  May be quick chat to explain would help ?

Reply:
I understand what it's for, but my point is that for someone who doesn't know what this function is doing, `foundArrival` doesn't really make sense. `arrivalToMerge` describes what the parameter is for